app/workers/new_page_worker.rb

```diff
@@ -0,0 +1,44 @@
+class NewPageWorker
+  include Sidekiq::Worker
+  attr_reader :page, :doc_html
+
+  def perform(page_id)
+    @page = Page.find(page_id)
+    @doc_html = download_content
+    process
+  rescue SocketError, Gastly::FetchError, OpenURI::HTTPError, Net::OpenTimeout
+    page.update_attributes(status: Page::ERROR_STATUS)
+  end
+
+  private
+
+  def process
+    page.update_attributes(status: Page::IN_PROGRESS_STATUS)
+    parse_html_content
+    make_screenshot
+    page.update_attributes(status: Page::PROCESSED_STATUS)
+  end
+
+  def download_screenshot(file)
+    File.new(Gastly.capture(page.source_url, file.path))
+  end
+
+  def make_screenshot
+    file = Tempfile.new
+    page.update_attributes(screenshot: download_screenshot(file))
+    file.unlink
+  end
+
+  def download_content
+    file = open(page.source_url)
+    Nokogiri::HTML(file)
+  end
+
+  def parse_html_content
+    unless doc_html.content == ''
```

Comment:
change to "if xxxx.present?"

try to use "if" always (or as much as possible)

Reply:
fyi,
active support provides 2 really cool methods `#present?` and `#blank?`, i'd suggest to use those